crates/astria-conductor/src/block_verifier.rs

```diff
@@ -193,8 +195,9 @@ fn validate_sequencer_namespace_data(
     let Some(data_hash) = header.data_hash else {
         bail!("data hash should not be empty");
     };
+    let action_tree_root_hash = sha2::Sha256::digest(action_tree_root);
     action_tree_root_inclusion_proof
-        .verify(action_tree_root, data_hash)
+        .verify(&action_tree_root_hash, data_hash)
```

Comment:
Somehow this is very surprising to me. As a user, my expectation is that I can pass in the action_tree_root as-is without needing to further manipulate it.

After all, isn't the root just the recursive hash of the tree?

Reply:
copy pasting my response in slack:

check this function here: https://github.com/astriaorg/astria/blob/main/crates/astria-sequencer-types/src/sequencer_block_data.rs#L363 this is used in SequencerBlockData::from_tendermint_block

the verification that was broken was the inclusion of the action_tree_root (which is currently the first tx in a block) inside data_hash (which is the root of the tree where the leaves are the transaction hashes)

 since the input to create the tree is the transaction hashes and not the transactions, and action_tree_root is a transaction, we need to hash it before verifying it was included in the data_hash tree